docs/anchoring.md

```diff
@@ -0,0 +1,212 @@
+it's good to understand what this string encodes.  
+By CIDv1 format defined at https://github.com/multiformats/cid scheme of CID is:
+```
+<hr-cid> ::= <hr-mbc> "-" <hr-cid-mc> "-" <hr-mc> "-" <hr-mh>
+```
+- `<hr-mbc>` is a human-readable multibase code (eg `base58btc`)
+- `<hr-cid-mc>` is the string `cidv#` (eg `cidv1` or `cidv2`)
+- `<hr-mc>` is a human-readable multicodec code (eg `cbor`)
+- `<hr-mh>` is a human-readable multihash (eg `sha2-256-256-abcdef0123456789...`)
+
+For example:
+
+```
+# example CID
+zb2rhe5P4gXftAwvA4eXQ5HJwsER2owDyS9sKaQRRVQPn93bA
+# corresponding human readable CID
+base58btc - cidv1 - raw - sha2-256-256-6e6ff7950a36187a801613426e858dce686cd7d7e3c0fc42ee0330072d245c95
+```
+
+Be more straightforward the CID is standard presentation of different types of hash values (calculated by various hash functions).
+And this hash calculates from initial document content (for `docId` in request body, it's immutable for this document lifetime) and
+changes (diffs) were made to this initial document (for `cid` in request body).
+
+
+If anchoring service received multiple requests for the same
+`docId` in anchoring interval then only last request will be anchored, others get 
+status `OUTDATED`.
+
+For more complete understanding of Anchoring working principle, it require to consider
+Merkle trees and Merkle DAGs.
+
+[Merkle tree](https://en.wikipedia.org/wiki/Merkle_tree) is an abstraction and data structure
+ that's commonly using in blockchain industry. This structure allow us to compress
+ big hash sequence into more compact one with possibility to check whether some hash was in initial 
+ sequence or not.
+ 
+![Merkle tree example schema](assets/banana_tree.png "Merkle tree example schema") 
+
+<small>Image derived from: https://media.consensys.net/ever-wonder-how-merkle-trees-work-c2f8b7100ed3</small>
+ 
+ There are several ways to hash data for Merkle tree at all. 
+ In this particular Ethereum-based case `keccak256` (modified version of SHA-3 final release) hash function is used.
+ 
+ DAGs stand for "Directed Acyclic Graphs", in other words - directed graph with no directed cycles, 
+ there is no way to return in some vertex once we start from it.
+ 
+ IPFS use Merkle DAGs instead of pure Merkle trees, the differences between them are:
+ * DAGs does not requires to be balanced
+ * Each DAG node can contain a value (payload)
+ * Each DAG node can have multiple parent nodes
+ 
+ When anchoring service handles new anchoring request the next actions occurs:
+ - Merkle tree from incoming requests is created, taking `cid` field of requests as Merkle leaves values.
+ - Merkle Root from Merkle tree is taken.
+ - put Merkle root on blockchain using hdwallet configured from mnemonic in `.env` file
+ - put anchor proof to IPFS using next [IPDL](https://ipld.io) structure and getting `proofCid` as a result:
+ ```
+{
```

Comment:
This is not IPDL structure. This is a raw excerpt from the code. Please, specify `the structure`.

Reply:
Where is path here?